modestpy/estim/ga_parallel/ga_parallel.py

```diff
@@ -0,0 +1,442 @@
+
+        # Reset temp placeholder
+        MODESTGA.TMP_SUMMARY = pd.DataFrame(columns=self.summary_cols)
+
+        # Return DataFrame with estimates
+        par_vec = outx
+        par_df = pd.DataFrame(columns=[x.name for x in self.est], index=[0])
+        for col, x in zip(par_df.columns, par_vec):
+            par_df[col] = x
+
+        return par_df
+
+    @staticmethod
+    def scale(v, lo, hi):
+        # scaled = (rescaled - lo) / (hi - lo)
+        return (v - lo) / (hi - lo)
+
+    @staticmethod
+    def rescale(v, lo, hi):
+        # rescaled = lo + scaled * (hi - lo)
+        return lo + v * (hi - lo)
+
+    def get_plots(self):
+        """
+        Returns a list with important plots produced by this estimation method.
+        Each list element is a dictionary with keys 'name' and 'axes'. The name
+        should be given as a string, while axes as matplotlib.Axes instance.
+
+        :return: list(dict)
+        """
+        plots = list()
+        plots.append({'name': 'MODESTGA',
+                      'axes': self.plot_parameter_evo()})
+        return plots
+
+    def save_plots(self, workdir):
+        self.plot_comparison(os.path.join(workdir, 'ps_comparison.png'))
+        self.plot_error_evo(os.path.join(workdir, 'ps_error_evo.png'))
+        self.plot_parameter_evo(os.path.join(workdir, 'ps_param_evo.png'))
+
+    def plot_comparison(self, file=None):
+        return plots.plot_comparison(self.res, self.ideal, file)
+
+    def plot_error_evo(self, file=None):
+        err_df = pd.DataFrame(self.summary[MODESTGA.ERR])
+        return plots.plot_error_evo(err_df, file)
+
+    def plot_parameter_evo(self, file=None):
+        par_df = self.summary.drop([MODESTGA.METHOD], axis=1)
+        par_df = par_df.rename(columns={
+            x: 'error' if x == MODESTGA.ERR else x for x in par_df.columns
+            })
+
+        # Get axes
+        axes = par_df.plot(subplots=True)
+        fig = figures.get_figure(axes)
+        # x label
+        axes[-1].set_xlabel('Iteration')
+        # ylim for error
+        axes[-1].set_ylim(0, None)
+
+        if file:
+            fig.set_size_inches(MODESTGA.FIG_SIZE)
+            fig.savefig(file, dpi=MODESTGA.FIG_DPI)
+        return axes
+
+    def get_full_solution_trajectory(self):
+        """
+        Returns all parameters and errors from all iterations.
+        The returned DataFrame contains columns with parameter names,
+        additional column '_error_' for the error and the index
+        named '_iter_'.
+
+        :return: DataFrame
+        """
+        return self.summary
+
+    def get_error(self):
+        """
+        :return: float, last error
+        """
+        return float(self.summary[MODESTGA.ERR].iloc[-1])
+
+    def get_errors(self):
+        """
+        :return: list, all errors from all iterations
+        """
+        return self.summary[MODESTGA.ERR].tolist()
+
+    # PRIVATE METHODS
+
+    @staticmethod
```

Comment:
The `_callback()` method is written as a static method to underline that it's a part of MODESTGA and useful only for this class.
Technically it could be a standalone function in a standalone module, but that would suggest it could be useful for other classes and modules as well, while it isn't.